### PR TITLE
SWMAAS-734 Refactor Walk and Voice scenarios to extend Routing scenario

### PR DESCRIPTION
**Pull request by npike**
For https://jira.phunware.com/browse/SWMAAS-728 the ask from ML has been to make sure “voice” and “walking” scenarios have feature parity with iOS.  Namely that they have:
- Route Summary
- Walking time

The activities that represent these two scenarios share maybe 90% of their code with the activity for Routing scenario — so I have updated Routing Scenarios Activity to be a little cleaner, and these two scenarios will extend it and decorate it with their Scenario specific logic

So we end up with 3 activities:
- RoutingActivity - has all of the logic for loading a building, picking POIs, and starting a route
- WalkTimeActivity - extends RoutingActivity, adds walking stuff
- VoicePromptActivity - extends WalkTimeActivity, adds text-to-speech stuff

This refactor is largely to avoid all 3 scenarios from implementing the same things three different times (map loading, walking time, route summary).

**This PR has no visual changes yet (except for AudioPromp Scenario now correctly showing walking calculations).  Just refactoring.**

**Notes**
- routing.xml is now activity_routing.xml
- activity_routing.xml will contain all of the shared elements across scenarios, just hidden unless a scenario turns them on
- I have updated activity_routing.xml to address incorrect textsizes, lint warnings, and have converted its parent layout from a RelativeLayout to a ConstraintLayout to match best practices.

Routing Scenario

Walk Time Scenario

Voice Prompt Scenario

### CODE REVIEW

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/OneWayRoutingActivity.kt [hunk 1]
```diff
@@ -112,7 +117,7 @@ class OneWayRoutingActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
 
         // Create the map manager and fragment used to load the building
         mapManager = PhunwareMapManager.create(this)
-        val mapFragment = fragmentManager.findFragmentById(R.id.map) as MapFragment
+        val mapFragment = supportFragmentManager.findFragmentById(R.id.map) as SupportMapFragment
```
Comment (npike) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/OneWayRoutingActivity.kt]: We should never being using Platform/Native fragments.  Where possible we and our customers should only be using Support Fragments.  Thankfully we provide a "support" version of our MapFragment -- so I am using this here.

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt [hunk 2]
```diff
@@ -1,63 +1,24 @@
...
-import com.phunware.mapping.manager.Router
-import com.phunware.mapping.model.*
-import java.lang.ref.WeakReference
-import java.util.ArrayList
+import com.phunware.mapping.model.RouteOptions
 import java.util.Locale
 
-class VoicePromptActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
-        Building.OnFloorChangedListener, Navigator.OnManeuverChangedListener, TextToSpeech.OnInitListener {
+class VoicePromptActivity : WalkTimeActivity(), TextToSpeech.OnInitListener {
```
Comment (npike) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt]: VoicePromptActivity now extends WalkTimeActivity, so we can remove all of the map loading, routing, and floor switching logic.  VoicePromptActivity largely now just contains code demonstrating the scenario.

File: Samples/kotlin/src/main/res/layout/activity_routing.xml [hunk 3]
```diff
@@ -1,28 +1,63 @@
...
     android:layout_height="match_parent">
 
-    <fragment xmlns:android="http://schemas.android.com/apk/res/android"
+    <fragment
         android:id="@+id/map"
-        android:name="com.phunware.mapping.MapFragment"
+        android:name="com.phunware.mapping.SupportMapFragment"
         android:layout_width="match_parent"
-        android:layout_height="match_parent" />
+        android:layout_height="match_parent"
```
Comment (dtran-phunware) [inline Samples/kotlin/src/main/res/layout/activity_routing.xml]: is match_parent advised when all 4 edges are constrained to parent? I thought we were to use 0dp or something.

Comment (npike) [reply Samples/kotlin/src/main/res/layout/activity_routing.xml]: Great catch.  Will fix.

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/RoutingActivity.kt [hunk 4]
```diff
@@ -205,6 +262,7 @@ class RoutingActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
     }
 
     private fun showRoutingDialog() {
+        // TODO this should be a DialogFragment
         val builder = AlertDialog.Builder(this)
 
         // Load custom dialog view
```
Comment (dtran-phunware) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/RoutingActivity.kt]: is it possible to do something about this inflate call passing null for root?

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt [hunk 5]
```diff
@@ -66,60 +27,23 @@ class VoicePromptActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
...
-            stopNavigating()
-        } else {
-            super.onBackPressed()
-        }
+        val sharedPref = getPreferences(Context.MODE_PRIVATE)
+        voiceEnabled = sharedPref.getBoolean(PREF_VOICE_ENABLED, false)
     }
 
     public override fun onDestroy() {
         if (tts != null) {
```
Comment (dtran-phunware) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt]: Is this null check necessary?

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt [hunk 6]
```diff
@@ -414,22 +115,22 @@ class VoicePromptActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
...
-            navigator?.let {navigator ->
+            navigator?.let { navigator ->
                 currentManeuverPosition?.apply {
                     textToVoice(getTextForPosition(navigator, this))
                 }
             }
         } else {
             voice.setImageResource(R.drawable.ic_muted)
             voiceStatusTextView.setText(R.string.demo_voice_prompt_muted)
-            tts!!.stop();
```
Comment (dtran-phunware) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/VoicePromptActivity.kt]: lol

**Comment by npike** [comment]
@dtran-phunware updated PR based on feedback, and refactored the dialog used for routing POI selection to be a proper DialogFragment.

**Pull request merged**